packages/react-components/src/components/graph/graph.tsx

```diff
@@ -1,40 +1,78 @@
-import React, { FC, HTMLAttributes, useCallback, useEffect, useRef } from 'react';
-import type { Core, ElementDefinition, LayoutOptions } from 'cytoscape';
-import { FitIcon, MinusIcon, PlusIcon, TargetIcon } from './lib/svgs/icons';
-import Container from '@awsui/components-react/container';
+import React, { HTMLAttributes, useCallback, useEffect, useRef } from 'react';
+import type { Core, ElementDefinition, EventObject } from 'cytoscape';
+import { response3 as response } from './lib/response';
 
-import CytoscapeComponent from 'react-cytoscapejs';
+import useCyEvent from './hooks/useCyEvent';
+import GraphView from './graph-view';
+import Toolbar from './graph-toolbar';
 
 import './styles.scss';
+import { Button } from '@cloudscape-design/components';
+import { STYLE_PREFIX } from './constants';
+import useStylesheet from './cytoscape-cloudscape-theme';
 
-export interface CytoGraphProps extends HTMLAttributes<HTMLDivElement> {
-  graphLayout?: LayoutOptions;
+export interface GraphProps extends HTMLAttributes<HTMLDivElement> {
+  onNodeSelected?: (e: EventObject) => void;
+  onEdgeSelected?: (e: EventObject) => void;
 }
 
 export const ZOOM_INTERVAL = 1;
 
-export const Graph: FC<CytoGraphProps> = ({ graphLayout, className, ...props }) => {
-  const elements = [
-    { data: { id: 'one', label: 'Node 1' }, position: { x: 0, y: 0 } },
-    { data: { id: 'two', label: 'Node 2' }, position: { x: 100, y: 0 } },
-    { data: { source: 'one', target: 'two', label: 'Edge from Node1 to Node2' } },
-  ];
+export type Node = {
+  entityId: string;
+  entityName: string;
+};
+
+export type Edge = {
+  sourceEntityId: string;
+  targetEntityId: string;
+  relationshipName: string;
+};
+
+export function Graph({ className, onNodeSelected, onEdgeSelected, ...props }: GraphProps) {
+  // TODO: Refactor this to use the datasource + query to get the data.
+  const elements: ElementDefinition[] = response.rows
+    .map((r) => r.rowData)
+    .reduce((acc, rowData) => {
+      const newElements = response.columnDescriptions.map((col, i) => {
+        const entity = rowData[i];
+        if (col.type === 'NODE') {
+          const node = entity as Node;
+          return { data: { id: node.entityId, label: node.entityName, ...node } } as ElementDefinition;
+        } else {
+          const relationship = entity as Edge;
```

Comment:
Nit: These areas of casting `entity` may be improved with type predicates for enhanced type safety.